Full-Stack-Collective/connectient
# Create popup for Admin Dashboard to show appointment details, schedule & cancel the appointment

## Description

As part of this PR, I've:

- [ ] Created a popup that opens on clicking individual appointment rows. The popup has:

- A tab to show Appointment details of the selected appointment.
- A tab to schedule the appointment - date and time
- A tab to cancel the appointment

- [ ] Incorporated features are:

- The **View Details** tab of the popup displays badges to indicate the status of the appointment, that updates in realtime if the appointment is scheduled or cancelled.
-  In the **View Details** tab of the popup, the email and contact numbers are links so when user clicks on it, they can be used to facetime or email.
- Right now,  in the **View Details** tab of the popup, the **Who scheduled the appointment?** is not getting any value. Should cover in future issues.  
- You can schedule an emergency and normal appointments for future dates and timing between 9am to 5pm. By default, it gives you a pre-filled form with 2 business days ahead and 9am as the suggested appointment dates and timings. The Sundays are disabled. 
- Right now, the scheduling form always shows all the timings. We need to take care of this issue where the form should consider the already booked time slots for a particular day.
- Once an appointment is scheduled or cancelled, the details are reflected in realtime on the popup and also, in the dashboard such that it moves the appointment to the `Scheduled` or `Cancelled` status.
- Once an appointment is scheduled and the popup is opened again, the scheduling form prefills the scheduled date and time and the option changes to `Reschedule appointment`.
- Once an appointment is cancelled, the options to  `Schedule Appointment`and `Uncancel` are disabled.If a patient has cancelled an appointment, basically, he/she has to request a new appointment.
- Scheduling the appointment triggers a confirmation email to the patient.

## Related Issue

Closes #72

## Acceptance Criteria

Allow admin to update few other fields while confirming the appointment with patient:

- [ ] Scheduled Date
- [ ] Scheduled Time
- [ ] Whether the appointment is cancelled or not

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates
### After
<img width="580" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/044b5d1a-a080-45d6-812f-8ec2ac65d2fd">
<img width="578" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/feba9b92-cded-4b59-aebf-7f97e969de98">
<img width="569" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/8f820608-8368-4ec6-a512-47ab52d26a21">
<img width="575" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/aac0780e-c467-4605-b5d2-982288ffa717">
<img width="568" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/b6509e6e-d2ff-4c32-ab66-3ea694091b25">
<img width="557" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/7df078e4-3066-4798-af26-8c81b7038f40">

## Testing Steps / QA Criteria
Install the dependencies before starting the app to make sure you have all the newly added components.
